presto-parser/src/test/java/io/prestosql/sql/parser/TestSqlParserErrorHandling.java

```diff
@@ -112,7 +112,7 @@
                 {"SELECT a AS z FROM t WHERE x = 1 + ",
                         "line 1:36: mismatched input '<EOF>'. Expecting: <expression>"},
                 {"SELECT a AS z FROM t WHERE a. ",
-                        "line 1:29: mismatched input '.'. Expecting: '%', '*', '+', '-', '/', 'AT', '||', <expression>"},
+                        "line 1:29: mismatched input '.'. Expecting: '%', '*', '+', '-', '/', 'AND', 'AT', 'EXCEPT', 'FETCH', 'GROUP', 'HAVING', 'INTERSECT', 'LIMIT', 'OFFSET', 'OR', 'ORDER', 'UNION', '||', <EOF>, <predicate>"},
```

Comment:
I wonder why the `.` is the problem. It could be saying:
```
unexpected EOF, expecting: <identifier> (field)
```

Reply:
That's a good question. Let me investigate.

Reply:
It's because Antlr bails out and leaves the the parser at a state past `>primaryExpression>`, which is the one that recognizes `<identifier> . <identifier>`. I guess it's matching `<identifier>` greedily as a `<primaryExpression>' because that's all that it can match given the available tokens in the stream and then it says "oops, what do I do with this `.`" now?